x/wasm/types/types.go

```diff
@@ -429,3 +429,18 @@ func (a AccessConfig) AllAuthorizedAddresses() []string {
 	}
 	return []string{}
 }
+
+type TxContracts map[string]struct{}
+
+func NewTxContracts() TxContracts {
+	return make(TxContracts, 0)
+}
+
+func (a TxContracts) AddContract(checksum string) {
+	a[checksum] = struct{}{}
+}
+
+func (a TxContracts) Exists(checksum string) bool {
```

Comment:
I would make those arguments of type `[]byte` and do the conversion to string internally. This can then be one of:
- string() cast
- hex encoding
- base64 encoding
- Copy to array `[32]byte`

No need to change the encoding right now but good to have options later on